starfish/pipeline/features/spots/detector/gaussian.py

```diff
@@ -38,16 +42,16 @@ def __init__(
             than thresh are ignored. Reduce this to detect blobs with less
             intensities.
         overlap : float [0, 1]
-            If two spots have more than this fraction of overlap, the spots are combined (default = 0.5)
-        blobs_stack : Union[ImageStack, str]
-            ImageStack or the path or URL that references the ImageStack that contains the blobs.
+            If two spots have more than this fraction of overlap, the spots are combined
+            (default = 0.5)
         measurement_type : str ['max', 'mean']
             name of the function used to calculate the intensity for each identified spot area
 
         Notes
         -----
-        This spot detector is very sensitive to the threshold that is selected, and the threshold is defined as an
-        absolute value -- therefore it must be adjusted depending on the datatype of the passed image.
+        This spot detector is very sensitive to the threshold that is selected, and the threshold
```

Comment:
I think Cellprofiler has a good solution to this problem. Internally, it takes the analogue of blobs_stack, and scales it to lie in [0, 1]. As such, your threshold is now a number between [0, 1] as well -- which makes understanding this parameter across varying dynamic ranges more interpretable. I propose we follow this precedent. We either check whether or not blobs_stack is in [0,1] (and threshold as well) and throw an exception if not. OR we simply do the transformation ourselves on blobs_stack and throw an exception if threshold is not in [0,1]. What do you think?

Reply:
I agree. I think this change would ideally follow after Tony's solution for dtypes, since it sounds like the goal here would be consistent. I expect he'll probably write a few basic functions to check that the images are in the right dtype, and that we could leverage them here.